python/lsst/ts/watcher/mock_squadcast.py

```diff
@@ -0,0 +1,229 @@
+      incident, but the manual says it is required. This emulates the manual.
+
+    Attributes
+    ----------
+    port : `int`
+        If the specified port is 0, then the chosen port
+        (after `start` has run). Otherwise the specified port.
+        The default is 80 because that is what OpsGenie uses.
+    url : `str`
+        The root URL of the service. "" until `start` has run.
+    endpoint_url : `str`
+        The URL of the endpoint (including the secret API key).
+        "" until `start` has run.
+    """
+
+    def __init__(self, port: int = 80) -> None:
+        self.port = port
+        self.escalation_key = os.environ["ESCALATION_KEY"]
+        # Dict of alert ID: data_dict
+        self.incidents: dict[str, dict] = dict()
+        self.runner: web.AppRunner | None = None
+        self.site: web.TCPSite | None = None
+        self.url: str = ""
+        self.endpoint_url = ""
+        # Unit tests may set this True to reject the next request
+        self.reject_next_request = False
+
+    def make_app(self) -> web.Application:
+        """Make an instance of the web application."""
+        app = web.Application()
+        url_suffix = INCIDENT_WEBHOOK_URL_SUFFIX + self.escalation_key
+        app.add_routes([web.post(url_suffix, self.incident_webhook)])
+        return app
+
+    async def start(self) -> None:
+        """Start the service.
+
+        Raises
+        ------
+        RuntimeError
+            If port = 0 and serving on more than one socket
+            (in which case the served port cannot be determined).
+        """
+        if self.runner is not None:
+            raise RuntimeError("Already started")
+        app = self.make_app()
+        self.runner = web.AppRunner(app)
+        await self.runner.setup()
+        self.site = web.TCPSite(self.runner, "127.0.0.1", port=self.port)
+        await self.site.start()
+        server = self.site._server
+        if self.port == 0:
+            if len(server.sockets) != 1:
+                raise RuntimeError(
+                    "Serving on more than one socket; cannot determine the port"
+                )
+            self.port = server.sockets[0].getsockname()[1]  # type: ignore
+        self.url = f"http://127.0.0.1:{self.port}"
+        self.endpoint_url = self.url + INCIDENT_WEBHOOK_URL_SUFFIX + self.escalation_key
+
+    async def close(self) -> None:
+        """Stop the service."""
+        if self.runner is None:
+            raise RuntimeError("No runner; did you call start?")
+        await self.runner.cleanup()
+
+    async def incident_webhook(self, request: web.Request) -> web.json_response:
+        """Request handler to trigger or resolve an incident.
+
+        Parameters
+        ----------
+        request : `aiohttp.web.Request`
+            Request.
+
+        Raises
+        ------
+        `web.HTTPRequestEntityTooLarge`
+            If the request is too large.
+        `web.HTTPUnauthorized`
```

Comment:
Several HTTPBadRequests get raised but not HTTPUnauthorized. Please amend the doc string accordingly.

Reply:
Thanks! That was a holdover from when I thought an authorization header was required (the vendor's docs are not very thorough). It definitely needs to be deleted.